granary/flickr.py

```diff
@@ -0,0 +1,371 @@
+              'username': person.get('username'),
+              'id': self.tag_uri(person.get('nsid')),
+              'image': {
+                'url': self.get_user_image(person.get('iconfarm'),
+                                           person.get('iconserver'),
+                                           person.get('nsid')),
+              },
+            },
+            'created': self.reformat_unix_time(activity.get('favedate')),
+            'url': '{}#liked-by-{}'.format(
+              activity.get('url'), person.get('nsid')),
+            'object': {'url': activity.get('url')},
+            'id': self.tag_uri('{}_liked_by_{}'.format(
+              photo.get('id'), person.get('nsid'))),
+            'objectType': 'activity',
+            'verb': 'like',
+          })
+
+      result['items'].append(activity)
+
+    return util.trim_nulls(result)
+
+  def get_actor(self, user_id=None):
+    """Get an ActivityStreams object of type 'person' given a Flickr user's nsid.
+    If no user_id is provided, this method will make another API requeset to
+    find out the currently logged in user's id.
+
+    Args:
+      user_id: string, optional
+
+    Returns:
+      dict, an ActivityStreams object
+    """
+    if not user_id:
+      user_id = self.user_id
+
+    if not user_id:
+      resp = self.call_api_method('flickr.people.getLimits')
+      user_id = resp.get('person', {}).get('nsid')
+
+    logging.debug('calling flickr.people.getInfo with user_id %s', user_id)
+    resp = self.call_api_method('flickr.people.getInfo', {'user_id': user_id})
+    return self.user_to_actor(resp)
+
+  def user_to_actor(self, resp):
+    """Convert a Flickr user dict into an ActivityStreams actor.
+    """
+    person = resp.get('person', {})
+    username = person.get('username', {}).get('_content')
+    obj = util.trim_nulls({
+      'objectType': 'person',
+      'displayName': person.get('realname', {}).get('_content') or username,
+      'image': {
+        'url': self.get_user_image(person.get('iconfarm'),
+                                   person.get('iconserver'),
+                                   person.get('nsid')),
+      },
+      'id': self.tag_uri(username),
+      # numeric_id is our own custom field that always has the source's numeric
+      # user id, if available.
+      'numeric_id': person.get('nsid'),
+      'location': {
+        'displayName': person.get('location', {}).get('_content'),
+      },
+      'username': username,
+      'description': person.get('description', {}).get('_content'),
+    })
+
+    # fetch profile page to get url(s)
+    profile_url = person.get('profileurl', {}).get('_content')
+    if profile_url:
+      try:
+        logging.debug('fetching flickr profile page %s', profile_url)
+        resp = urllib2.urlopen(
+          profile_url, timeout=appengine_config.HTTP_TIMEOUT)
+        profile_json = mf2py.parse(doc=resp, url=profile_url)
+        # personal site is likely the first non-flickr url
```

Comment:
no unfortunately. it's just a list of urls they are free to enter; I'm just assuming most people put their "primary representative" site first, instead of like their employer